src/Streamly/Coreutils/Stat.hs

```diff
@@ -0,0 +1,38 @@
+-- |
+-- Module      : Streamly.Coreutils.Stat
+-- Copyright   : (c) 2022 Composewell Technologies
+-- License     : Apache-2.0
+-- Maintainer  : streamly@composewell.com
+-- Stability   : experimental
+-- Portability : GHC
+--
+-- gets the FileStatus information (user ID,
+-- size, access times, etc.) for the file path.
+
+module Streamly.Coreutils.Stat
+    ( Stat
+    , stat
+    , deRef
+    , file
+    ) where
+
+import System.Posix.Files (FileStatus)
+import qualified System.Posix.Files as Files
+import Data.Default.Class (Default (def))
+
+data Stat = DeRef | File
+
+deRef :: Stat
+deRef = DeRef
+
+file :: Stat
+file = File
```

Comment:
Instead of using two options `deRef` and `file`, just use one called `followLinks :: Switch -> Stat -> Stat` and use the default value as `Off`.